src/utils.js

```diff
@@ -261,11 +305,13 @@ let toolpath = global.packageInfo.package
     )
   : path.join(__dirname, "..", "platform-tools", platforms[os.platform()]);
 let processes = [];
+
+// AW : Modifying the maxBuffer size to handle the output of the backup (all file path backuped on the device!)
 function execTool(tool, args, callback) {
   let pid = exec(
     [path.join(toolpath, tool)].concat(args).join(" "),
     {
-      maxBuffer: 1024 * 1024 * 2
+      maxBuffer: 1024 * 1024 * 100
```

Comment:
Not a good idea to do it like this. It'd be better to tar on the device and adb pull.

Reply:
For now I leave it like this, I will take a look on how to handle this in a better way.

Reply:
Makes sense, especially since you need it to continue testing your current state over at your promise-android-tools pr :)